File: engagelens_app/lib/models/attendance.dart
/// models/attendance.dart
class AttendanceRecord {
  final String studentId;
  final String name;
  final String rollNo;
  final String classSection;
  final String date;
  final String session;
  final String timestamp;
  final String status;
  final String matchedAngle;
  final double matchDistance;

  AttendanceRecord({
    required this.studentId,
    required this.name,
    this.rollNo = '',
    this.classSection = '',
    required this.date,
    required this.session,
    this.timestamp = '',
    this.status = 'Present',
    this.matchedAngle = '',
    this.matchDistance = 0.0,
  });

  factory AttendanceRecord.fromJson(Map<String, dynamic> json) {
    return AttendanceRecord(
      studentId: json['student_id'] as String,
      name: json['name'] as String,
      rollNo: (json['roll_no'] as String?) ?? '',
      classSection: (json['class_section'] as String?) ?? '',
      date: json['date'] as String,
      session: json['session'] as String,
      timestamp: (json['timestamp'] as String?) ?? '',
      status: (json['status'] as String?) ?? 'Present',
      matchedAngle: (json['matched_angle'] as String?) ?? '',
      matchDistance: ((json['match_distance'] as num?) ?? 0).toDouble(),
    );
  }
}

File: engagelens-api/models/attendance.py
# Pydantic models for attendance endpoints
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel


class AttendanceMarkRequest(BaseModel):
    student_id: str
    name: str
    matched_angle: str = "front"
    match_distance: float = 0.0
    session: str = "FN"
    roll_no: str = ""
    class_section: str = ""


class AttendanceRecord(BaseModel):
    student_id: str
    name: str
    roll_no: str = ""
    class_section: str = ""
    date: str
    session: str
    timestamp: str = ""
    status: str = "Present"
    matched_angle: str = ""
    match_distance: float = 0.0


class AttendanceStatsResponse(BaseModel):
    total_students: int
    total_records: int
    by_date: dict


class AbsenteeResponse(BaseModel):
    student_id: str
    name: str
    roll_no: str = ""
    class_section: str = ""
    parent_telegram_chat_id: Optional[str] = None
